driver-core/src/main/java/com/datastax/driver/core/TypeCodec.java

```diff
@@ -1096,26 +1115,49 @@ public Long parse(String value) {
 
         @Override
         public String format(Long value) {
```

Comment:
As I suggested for JAVA-817, the format() method now returns a valid, human-readable representation of the time value, instead of the number of nanoseconds since midnight.
We could do a similar thing when formatting timestamp types (currently we format them as milliseconds since the Epoch).